detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,246 @@
+var _ detectorUtils = (*eksDetectorUtils)(nil)
+
+// Detect detects associated resources when running with AWS EKS.
+func (detector *ResourceDetector) Detect(ctx context.Context) (*resource.Resource, error) {
+
+	labels := []label.KeyValue{}
+
+	isEks, err := isEks(detector.utils)
+	if hasProblem(err) {
+		return nil, err
+	}
+
+	if !isEks {
+		return resource.New(labels...), nil
+	}
+
+	clusterName, err := getClusterName(detector.utils)
+	if hasProblem(err) {
+		return nil, err
+	}
+	if clusterName != "" {
+		labels = append(labels, semconv.K8SClusterNameKey.String(clusterName))
+	}
+
+	containerID, err := detector.utils.getContainerID()
+	if hasProblem(err) {
+		return nil, err
+	}
+	if containerID != "" {
+		labels = append(labels, semconv.ContainerIDKey.String(containerID))
+	}
+
+	return resource.New(labels...), nil
+
+}
+
+func isEks(utils detectorUtils) (bool, error) {
+	if !isK8s(utils) {
+		return false, nil
+	}
+
+	awsAuth, err := utils.fetchString("GET", k8sSvcURL+authConfigmapPath)
+	if hasProblem(err) {
+		return false, err
+	}
+
+	return awsAuth != "", nil
+}
+
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// eksDetectorUtils is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	fmt.Println("REAL HIT")
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// fetchString is implementing the detectorUtils interface
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	authHeader, err := getK8sCredHeader()
+	if hasProblem(err) {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	client := &http.Client{
+		Timeout: millisecondTimeOut * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	response, err := client.Do(request)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getK8sCredHeader() (string, error) {
+	content, err := ioutil.ReadFile(k8sTokenPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	return "Bearer " + string(content), nil
+}
+
+func getClusterName(utils detectorUtils) (string, error) {
+	resp, err := utils.fetchString("GET", k8sSvcURL+cwConfigmapPath)
+	if hasProblem(err) {
+		return "", err
+	}
+
+	var parsedResp JSONResponse
+	err = json.Unmarshal([]byte(resp), &parsedResp)
+	if hasProblem(err) {
+		return "", err
+	}
```

Comment:
Building on my other comment on hasProblem, is logging this error enough for the user to know what happened?

The error will likely give no context about why you were unmarshaling JSON. Which is why you probably want to give a specific error log in each case, or add a way to add information in the call to hasProblem